crates/core/src/event.rs

```diff
@@ -13,13 +15,30 @@ use crate::ibc::IbcEvent;
 
 /// Used in sub-systems that may emit events.
 pub trait EmitEvents {
-    /// Emit an event
-    fn emit(&mut self, value: Event);
+    /// Emit a single [event](Event).
+    fn emit<E>(&mut self, event: E)
+    where
+        E: Into<Event>;
+
+    /// Emit a batch of [events](Event).
+    fn emit_many<B, E>(&mut self, event_batch: B)
```

Comment:
Since we have only two implementation of this trait, and one of the two overrides this default, I wonder if we should avoid this default here and maybe implement this one on `Vec<Event>` using `std::iter::Extend::extend`